lsp-sourcekit.el

```diff
@@ -41,31 +41,39 @@
 ;;   Customization
 ;; ---------------------------------------------------------------------
 
-(defcustom lsp-sourcekit-executable
-  "sourcekit"
-  "Path of the lsp-sourcekit executable."
+
+(defgroup lsp-sourcekit nil
+  "LSP support for Swift & C-family languages (C, C++,
+Objective-C, Objective-C++) using sourcekit-lsp."
+  :group 'lsp-mode
+  :tag "Language Server"
+  :link '(url-link "https://github.com/apple/sourcekit-lsp"))
+
+(defcustom lsp-sourcekit-executable "sourcekit-lsp"
+  "The sourcekit-lsp executable to use.
+Leave as just the executable name to use the default behavior of
+finding the executable with `exec-path'."
   :type 'file
-  :group 'sourcekit)
+  :group 'lsp-sourcekit)
 
-(defcustom lsp-sourcekit-extra-args
-  nil
-  "Additional command line options passed to the lsp-sourcekit executable."
+(defcustom lsp-sourcekit-extra-args nil
+  "Extra arguments for the sourcekit-lsp executable."
   :type '(repeat string)
-  :group 'sourcekit)
+  :group 'lsp-sourcekit)
 
 ;; ---------------------------------------------------------------------
 ;;  Register lsp client
 ;; ---------------------------------------------------------------------
 
 (defun lsp-sourcekit--lsp-command ()
   "Generate the language server startup command."
-  `(,lsp-sourcekit-executable
-    ,@lsp-sourcekit-extra-args))
+  `(,lsp-sourcekit-executable ,@lsp-sourcekit-extra-args))
 
 (lsp-register-client
- (make-lsp-client :new-connection (lsp-stdio-connection 'lsp-sourcekit--lsp-command)
-                  :major-modes '(swift-mode)
-                  :server-id 'sourcekit-ls))
+ (make-lsp-client :new-connection (lsp-stdio-connection
+				   'lsp-sourcekit--lsp-command)
+                  :major-modes '(swift-mode c-mode c++-mode objc-mode)
```

Comment:
I'm thinking that, instead of using `:major-modes`, we could use `:activation-fn` with a custom function. Inside that custom function we can be more precise for when `sourcekit-lsp` should activate: 

Require that the major mode is either `swift-mode`, `c-mode`, `c++-mode`, or `objc-mode` AND that the project contains a `Package.swift` or `.xcodeproj`/`.xcworkspace` file. You can use Emacs's `locate-dominating-file` function to do this.

I think this will do the right thing in almost all cases, without the user having to configure existing or new projects that contain C++ code. What do you think?

Reply:
That sounds like the best option. Thanks for the pointers.